### PR TITLE
[Woo POS] Product list UI improvements

### LINKED ISSUE #12167: [Woo POS] M2: Add bottom padding to products list
Bottom padding should be added to the product list to avoid overlaying the last item in the list with the global floating menu:

Image

### PR DESCRIPTION
**Pull request by samiuelson**
Closes: #12167

### Description

This PR adds UI improvements to the product list, allowing the bottom element on the list to be fully visible. Adjusting the height of the bottom list item (Spacer) added the desired padding that caused the bottom list item to land above the floating toolbar.

### Steps to reproduce

Scroll the product list up and observe the last product in the list can always be visible.

### Images/gif

https://github.com/user-attachments/assets/cf59c31c-7942-4112-9207-75e5a9bd828e

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/products/WooPosProductsScreen.kt [hunk 1]
```diff
@@ -146,13 +146,11 @@ private fun WooPosProductsScreen(
                             onSimpleProductsBannerLearnMoreClicked,
                             onSimpleProductsBannerClosed
                         )
-                        Box {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/products/WooPosProductsScreen.kt:149]: The Box seems to be redundant.

**Review by malinajirka — APPROVED** [review]
Thanks @samiuelson for working on this!

I've tested this flow and it works as expected. However, the gap between the floating menu and the last item in the list seems off (too big). Here are screenshots from two devices I tested this on:




On your video, it looks fine since the bottom OS bar is transparent. I would consider using the same padding that we use between the edge of the app's drawing area and the bottom of the floating menu for this gap, wdyt?

Feel free to merge this PR after the change or if you believe it's better as is.

**Comment by samiuelson** [comment]
> On your video, it looks fine since the bottom OS bar is transparent. I would consider using the same padding that we use between the edge of the app's drawing area and the bottom of the floating menu for this gap, wdyt?


Thanks for review @malinajirka. I took the height (112dp) of the Spacer from Figma TfaZ4LUkEwEGrxfnEFzvJj-fi-2877_88678:

Screenshot 2024-08-06 at 12 56 51


You're right though, it turns out the actual height of the toolbar is off. 

I'm adjusting the height of the spacer to 104dp (866294c052bbc287cc31a33929700871213ec1b7). 104dp = height of the Toolbar + 24dp of bottom padding + 24dp of top padding.

This is how it looks on Samsung tablet in different modes:
| With task bar | Without task bar |
|---|---|
|Screenshot_20240806_125438|Screenshot_20240806_125552|

| Toolbar's height | Spacer height |
|---|---|
|Screenshot 2024-08-06 at 13 15 52|Screenshot 2024-08-06 at 13 15 30|

However, on your device, the bottom padding is much smaller. I don't know the reason for that. I'm going to merge the updated height for now, and we can improve it later.

**Pull request merged**